crates/chain/src/local_chain.rs

```diff
@@ -171,11 +171,16 @@ impl From<BTreeMap<u32, BlockHash>> for LocalChain {
 impl ChainOracle for LocalChain {
     type Error = Infallible;
 
-    fn is_block_in_chain(
+    fn is_block_in_chain<B: Into<BlockId>>(
         &self,
         block: BlockId,
-        chain_tip: BlockId,
+        chain_tip: Option<B>,
     ) -> Result<Option<bool>, Self::Error> {
+        let chain_tip: BlockId = match chain_tip {
+            Some(x) => x.into(),
+            None => return Ok(None),
+        };
+
```

Comment:
Wouldn't it better to return `Result<bool, Self::Error>`?
`Result<Option<bool>, Self::Error>` seems unnecessary

Reply:
There are three possibilities:

- `Some(true)`: The block is in the chain
- `Some(false)`: The block is not in the chain
- `None` we don't know if the block is in the chain.

Reply:
Maybe in the future, we can consider turning this into an enum?

Reply:
Not sure. Would just add documentation as far as I can see. Documenting the trait method seems clear enough.